modules/client/src/main/java/org/apache/ignite/jdbc/JdbcStatement.java

```diff
@@ -0,0 +1,647 @@
+
+        if (fetchSize <= 0)
+            throw new SQLException("Fetch size must be greater than zero.");
+
+        pageSize = fetchSize;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getFetchSize() throws SQLException {
+        ensureNotClosed();
+
+        return pageSize;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getResultSetConcurrency() throws SQLException {
+        ensureNotClosed();
+
+        return CONCUR_READ_ONLY;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int getResultSetType() throws SQLException {
+        ensureNotClosed();
+
+        return TYPE_FORWARD_ONLY;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void addBatch(String sql) throws SQLException {
+        ensureNotClosed();
+
+        batchSize++;
+
+        if (batch == null)
+            batch = new ArrayList<>();
+
+        batch.add(new JdbcQuery(sql, null));
+    }
+
+    /** {@inheritDoc} */
+    @Override public void clearBatch() throws SQLException {
+        ensureNotClosed();
+
+        batchSize = 0;
+
+        batch = null;
+    }
+
+    /** {@inheritDoc} */
+    @Override public int[] executeBatch() throws SQLException {
+        ensureNotClosed();
+
+        closeResults();
+
+        if (CollectionUtils.nullOrEmpty(batch))
+            return new int[0];
+
+        JdbcBatchExecuteRequest req = new JdbcBatchExecuteRequest(conn.getSchema(), batch, conn.getAutoCommit());
+
+        try {
+            JdbcBatchExecuteResult res = conn.handler().batch(req);
+
+            if (res.status() != JdbcResponse.STATUS_SUCCESS) {
+                throw new BatchUpdateException(res.err(),
+                    IgniteQueryErrorCode.codeToSqlState(res.status()),
+                    res.status(),
+                    res.updateCounts());
+            }
+
+            return res.updateCounts();
+        }
+        finally {
+            batchSize = 0;
+
+            batch = null;
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override public Connection getConnection() throws SQLException {
+        ensureNotClosed();
+
+        return conn;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean getMoreResults(int curr) throws SQLException {
+        ensureNotClosed();
+
+        if (resSets == null || curRes >= resSets.size())
+            return false;
+
+        curRes++;
+
+        if (resSets != null) {
+            assert curRes <= resSets.size() : "Invalid results state: [resultsCount=" + resSets.size() +
+                ", curRes=" + curRes + ']';
+
+            switch (curr) {
+                case CLOSE_CURRENT_RESULT:
+                    if (curRes > 0)
+                        resSets.get(curRes - 1).close0();
+
+                    break;
+
+                case CLOSE_ALL_RESULTS:
+                    for (int i = 0; i < curRes; ++i)
+                        resSets.get(i).close0();
+
+                    break;
+
+                case KEEP_CURRENT_RESULT:
+                    break;
```

Comment:
I'm not sure if we will support multiple open results.

>  *@exception SQLFeatureNotSupportedException if
     * <code>DatabaseMetaData.supportsMultipleOpenResults</code> returns
     * <code>false</code> and either
     *        <code>Statement.KEEP_CURRENT_RESULT</code> or
     *        <code>Statement.CLOSE_ALL_RESULTS</code> are supplied as
     * the argument.

Reply:
At the moment we do not support multiple open results. Now it throws SQLFeatureNotSupportedException.